mesonpy/_editable.py

```diff
@@ -317,12 +317,23 @@ def _rebuild(self) -> Node:
         env[MARKER] = os.pathsep.join((env.get(MARKER, ''), self._build_path))
 
         if self._verbose or bool(env.get(VERBOSE, '')):
-            print('+ ' + ' '.join(self._build_cmd))
-            stdout = None
+            # We do not want any output if there is no work to do. Code is adapted from:
+            # https://github.com/mesonbuild/meson/blob/a35d4d368a21f4b70afa3195da4d6292a649cb4c/mesonbuild/mtest.py#L1635-L1636
+            if self._build_cmd[0] == 'meson':
+                # On Windows meson compile is used so to do a dry run you need
+                # --ninja-args
+                dry_run_build_cmd = self._build_cmd + ['--ninja-args=-n']
+            else:
+                dry_run_build_cmd = self._build_cmd + ['-n']
+
+            p = subprocess.run(dry_run_build_cmd, cwd=self._build_path, env=env, capture_output=True, check=True)
+            if b'ninja: no work to do.' not in p.stdout and b'samu: nothing to do' not in p.stdout:
+                module_names = ' '.join(sorted(self._top_level_modules))
+                build_command = ' '.join(self._build_cmd)
+                print(f'meson-python: building {module_names} with {build_command!r}', flush=True)
+                subprocess.run(self._build_cmd, cwd=self._build_path, env=env)
```

Comment:
Why does Codecov think that these lines are not covered by the tests?

Reply:
Hmmm not sure what is happening ...

Reply:
It seems like a but somewhere in the toolchain (maybe `coverage`, maybe `codecov`, not sure ...) that does not see the line covered although the line is executed. 

- codecov says the line is not covered [build log](https://github.com/mesonbuild/meson-python/pull/579/checks?check_run_id=22255522195) or
![image](https://github.com/mesonbuild/meson-python/assets/1680079/9d62939c-a0a7-4054-8486-40610e0ec25c)

- the line is executed as you can see in this [build log](https://github.com/mesonbuild/meson-python/actions/runs/8143472943/job/22255402547?pr=579#step:9:204) for example

Reply:
Maybe sklearn's editable install `find_spec` is called prior to the initialization of the `coverage` tracer?

Reply:
> Maybe sklearn's editable install's

What does sklearn has to do with this?

Reply:
Sorry, I typed too quickly. I wanted to mention the editable modules of the tests.

Reply:
I have added a test to cover these lines in the main process, by adding a parametrized tests to `test_mesonpy_meta_finder` with `verbose=True`

> coverage (unless tricks are applied) does not cross process boundaries when collecting coverage data.

I have seen `parallel = True` in `.coveragerc` files to be able to cover lines in subprocesses, see [this](https://coverage.readthedocs.io/en/latest/cmd.html#combining-data-files-coverage-combine) for more details. Not sure if you would like to consider this option. We are doing this in scikit-learn and this seems to work fine but there may be some edge cases I am unaware of.